src/mixins/createStatefulChildrenMixin.ts

```diff
@@ -135,18 +188,52 @@ function manageChildrenState(evt: ChildListEvent<any, Child>) {
 	}
 }
 
-const createStatefulChildrenMixin: StatefulChildrenMixinFactory = createStateful
+const createStatefulChildrenMixin = compose({
+		createChild<C extends Child>(
+			this: StatefulChildren<Child, StatefulChildrenState>,
+			factory: ComposeFactory<C, any>,
+			options: any = {}
+		): Promise<[string, C]> {
+			if (managementMap.has(this)) {
+				const management = managementMap.get(this);
+				const { registry, id } = management;
+				if (!options.id) {
+					/* depending upon the construction lifecycle, the this.id may not have been properly set and will
+					 * auto-generate an ID, therefore we have copied the ID out of options, if it was present and will
+					 * use that as a base for autogenerating the child widget's ID */
+					options.id = `${id || this.id}-child-${++management.childrenUID}`;
+				}
+				return registry.create(factory, options)
+					.then(([ id, child ]): [ string, C ] => {
+						/* Because this mixin manages children in the state, we will append the child to the state
+						 * of the parent */
+						this.setState({ children: (this.state.children || []).concat([ id ]) });
```

Comment:
I chose something slightly different, because a) the contextual typing works at this code point and b) it avoids creating an unnecessary array when guarding against `undefined`.